### PR TITLE
[Feature] Emote menu search

### PR DESCRIPTION
**Pull request by ipiv**
Adds a text input to emote menu that filters visible emotes

Fixes night/betterttv#3841

### CODE REVIEW

File: src/templates/menu.html [hunk 1]
```diff
@@ -16,6 +16,7 @@
 			<div class="footer">
 				<a class="pull-left icon icon-home" href="http://cletusc.github.io/Userscript--Twitch-Chat-Emotes" target="_blank" title="Visit the homepage where you can donate, post a review, or contact the developer"></a>
 				<a class="pull-left icon icon-gear" data-command="toggle-editing" title="Toggle edit mode"></a>
+				<input type="text" placeholder="Search emotes" data-command="search-emotes">
```
Comment (night) [inline src/templates/menu.html:19]: i somehow missed this pr. this probably needs some styling to make it fit in better to the emote menu. can you propose some?

File: src/modules/ui.js [hunk 2]
```diff
@@ -457,6 +462,37 @@ UIMenu.prototype.getEmote = function (name) {
...
+	// Update visibility on emotes based on search
+	this.emotes.forEach(function (emote) {
+		// Skip if emote is marked hidden via Edit mode
+		if (!emote.isVisible()) {
+			return;
+		}
+
+		emote = self.getEmote(emote.getText());
+		var shouldShowEmote = emote.instance.getText().toLowerCase().includes(val);
+		emote.dom.toggleClass('emote-menu-hidden', !shouldShowEmote);
```
Comment (night) [inline src/modules/ui.js:485]: when the emote menu is closed and reopened, what is the state of the menu? does search reset? do emotes reset?

Comment (ipiv) [reply src/modules/ui.js:486]: Indeed the states weren't consistent, pushed a change that clears input on menu close.

File: src/templates/menu.html (hunk 1 above)
Comment (ipiv) [reply src/templates/menu.html]: Added some simple styling that should be fitting to the overall look.

**Pull request merged**